### ISSUE TITLE
Solve the AVATAR upload by the form, users are exchanging the photos and the old ones remain on the server.

### ISSUE THREAD

**Issue by ofmarconi** [CrocoBlock/suggestions#4724]
I'm using Profile Builder + Forms to upload Avatar image.

But the user can upload as many photos as he wants, being able to fill my server with unnecessary photos.

My request is:

Some way to delete photos not used by any user;

OR

Instead of uploading a new photo, the submission form replaces the current photo on the server as this plugin does on the backend: https://br.wordpress.org/plugins/enable-media-replace/

I know it may be impossible to attach how it works with posts:
https://www.facebook.com/groups/CrocoblockCommunity/posts/4196295160461873/?comment_id=5109099155848131&reply_comment_id=5119300328161347&__cft__[0]=AZVBqaZkiaJO8-GOiccafWiydfkEenXRYxoHj-QuuOakXxIqgLbT4G3CPHetTp7g_Jnf8YxNJHDPHD_A-wWCi62OO5c3eqS08863mzP8UKasYDD3xzuhGF4oj8T5hROjc76rUa55sd4xpuMKYoc9Bsj0&__tn__=R]-R

But it's something to think about...

========

An alternative solution would be to compress the images that are sent by the form, even if it is larger than 150px vs 150px, the form will cut the image to that size and compress the file, discarding the original. Is it possible to compress images sent from forms?

**Comment by nreljed** [comment]
yes , the current wordpress upload system has reached its limits. 😅

I didn't think it was possible...but here we are now

A new modern upload system is needed with a simple and mobile friendly design
I don't think the wordpress community is worried about this so we don't expect to see a change from wordpress team.

However 😃
crocoblocks with [jetengine](https://github.com/Crocoblock/suggestions/issues/3885) can introduce an answer to the question of: 

loss of control over uploads and their life on the server.

by dedicating a complete beta for this problem

thank god🙏😃  we already have an overview of the work of competitors

the solution quite naturally is to take what is best on 

🔎ACF field media
https://www.advancedcustomfields.com/resources/gallery/

- Library (see #3422)

- Minimum & Maximum (pixel size control)

🔎PODS field media
https://docs.pods.io/fields/file/

- Modal Title (the power to change the file title)

- Upload Directory (This allows you to change where uploads will be saved to on your site folder structure.)

🔎My ideas

- modern uploader design (see #3411)
- (option to choose) Define the first image as featured #3154
-  (option to choose) default attachement (include devtool plugin in core)
- (option to choose) absorb the idea of ​​[autoremove attachements](https://github.com/Crocoblock/jetformbuilder/issues/60) to eliminate files with no attachment 

Combine everything like a [filepond](https://pqina.nl/filepond/docs/api/server/) or write similar code from scratch.

and always keeping in mind that there will always be projects (except simple blog) who required this extensive control.

there will surely be a page or parameters entirely dedicated to the upload rules of one or more CPTs
real control over uploads and set strict rules for their validation and elimination backend or frontend using media/gallery field.

it will inevitably be the **best robust and modern system for uploading out of the box**
a real ++ and a correct answer to this problem 

there is a good chance that one day the devs will offer a **complete solution**

as projects have increasing upload needs and the server space is limited.
the current wordpres upload system is no longer meets our needs.😓
